coreneuron/nrniv/cellorder.cpp

```diff
@@ -81,13 +81,24 @@ InterleaveInfo& InterleaveInfo::operator=(const InterleaveInfo& info) {
 
 InterleaveInfo::~InterleaveInfo() {
     if (stride) {
+        #ifdef UNIFIED_MEMORY
         free_memory(stride);
         free_memory(firstnode);
         free_memory(lastnode);
         free_memory(cellsize);
+        #else
+        delete[] stride;
+        delete[] firstnode;
+        delete[] lastnode;
+        delete[] cellsize;
+        #endif
```

Comment:
I think we can avoid these if-else throughput the code. 

Instead of `delete` or `new`, we should specialise `free_memory` and  `ecalloc_align`  for `UNIFIED_MEMORY` or `NON_UNIFIED_MEMORY`.